src/filesystem/drive/inner.rs

```diff
@@ -48,11 +62,75 @@ impl InnerDrive {
             ))
     }
 
+    fn mark_ancestors_dirty(&mut self, node_id: NodeId) -> Result<(), OperationError> {
+        // Changes have happened in `node_id` walk up its parents to root marking nodes as dirty
+        let mut node_id = node_id;
+        self.dirty_nodes.push(node_id);
+        while let Some(parent_perm_id) = self.by_id(node_id)?.parent_id() {
+            // Setting a node's parent ID to its own ID is done in some operations such as moving a node
+            // to temporarily orphan it. Obviously this would cause an infinite loop and needs to be caught
+            if parent_perm_id == self.by_id(node_id)?.permanent_id() {
+                break;
+            }
```

Comment:
Hmmm this comment is concerning, no node should be its own parent and we shouldn't hold on to references for things we know don't exist (so if a node is removed it should have its ID removed from this if present). A movement operation should be able to locate its new parent and set that directly for example.

I'd consider this an internal corruption bug if the case ever occurs.

Reply:
So this is just working around the current behavior of the `DirectoryHandle::mv` which currently exhibits that behavior: https://github.com/banyancomputer/banyanfs/blob/main/src/filesystem/drive/directory_handle.rs#L283

I didn't take the time to dig in on exactly why that was done and instead just tried to accommodate it. I actually only discovered that behavior when I went to run my code and it got caught in an infinite busy loop. I'd be happy to look at changing that behavior though as part of this PR

Reply:
I think that's a bug in the implementation we shouldn't need to do that and my money is I was trying to set it to something else.

Reply:
Ok `mv` fix is in and this work around was removed